### PR TITLE
fix KeyboardAvoidingView for Files tab by plumbing through a headerHeight to nav2

### PR DESCRIPTION
**Pull request by songgao**
We are providing a custom header for the Files tab, and it breaks KeyboardAvoidingView. So set an explicit `keyboardVerticalOffset`.

Not sure if it's the right way to do it, but this works for me.

### CODE REVIEW

File: shared/fs/nav-header/mobile-header.js [hunk 1]
```diff
@@ -58,13 +58,19 @@ class MobileHeader extends React.PureComponent<Props, State> {
   }
 }
 
+const statusBarHeight = isIPhoneX ? 45 : 20
```
Comment (chrisnojima) [inline shared/fs/nav-header/mobile-header.js]: I believe this is in styles already

File: shared/router-v2/shim.native.js [hunk 2]
```diff
@@ -11,18 +11,23 @@ const shimNewRoute = (Original: any) => {
   class ShimmedNew extends React.PureComponent<any, void> {
     static navigationOptions = Original.navigationOptions
     render() {
+      const navigationOptions =
+        typeof Original.navigationOptions === 'function'
+          ? Original.navigationOptions(this.props.navigation)
+          : Original.navigationOpptions
```
Comment (chrisnojima) [inline shared/router-v2/shim.native.js]: Typo

File: shared/router-v2/shim.native.js [hunk 3]
```diff
@@ -11,18 +11,23 @@ const shimNewRoute = (Original: any) => {
   class ShimmedNew extends React.PureComponent<any, void> {
     static navigationOptions = Original.navigationOptions
     render() {
+      const navigationOptions =
+        typeof Original.navigationOptions === 'function'
+          ? Original.navigationOptions(this.props.navigation)
```
Comment (chrisnojima) [inline shared/router-v2/shim.native.js]: The params should be different. Should be an object.  See the nav docs

Comment (songgao) [reply shared/router-v2/shim.native.js]: Oh! that's why I had issue with navigation being undefined. Thanks!

File: shared/fs/nav-header/mobile-header.js (hunk 1 above)
Comment (songgao) [reply shared/fs/nav-header/mobile-header.js]: https://github.com/keybase/client/blob/3e6b7e6ebd9598f34d53c6d7c3034009b5368d0c/shared/styles/index.native.js#L51

doesn't seem to consider iPhoneX yet, so I added it to the new commit.

Comment (songgao) [reply shared/fs/nav-header/mobile-header.js]: (talked on chat; rebased over nojima's branch)

File: shared/router-v2/shim.native.js [hunk 4]
```diff
@@ -11,18 +11,23 @@ const shimNewRoute = (Original: any) => {
...
           behavior={Styles.isIOS ? 'padding' : undefined}
+          keyboardVerticalOffset={(navigationOptions && navigationOptions.headerHeight) || undefined}
         >
           {body}
         </Kb.NativeKeyboardAvoidingView>
       )
 
       // don't make safe areas
-      if (Original.navigationOptions && Original.navigationOptions.underNotch) {
+      if (navigationOptions && Original.navigationOptions.underNotch) {
```
Comment (chrisnojima) [inline shared/router-v2/shim.native.js]: should be navigationOptions.underNotch

**Pull request merged**